courses/react-js/advanced-level/redux-toolkit/lesson_1.md

```diff
@@ -0,0 +1,154 @@
+---
+id: lesson-1
+title: "Introduction to Redux Toolkit"
+sidebar_label: Lesson - 1
+sidebar_position: 1
+description: "In this lesson, you will learn the basics of Redux Toolkit, including its core concepts and how it simplifies the process of managing state in your React applications."
+tags:
+  [
+    courses,
+    redux-toolkit,
+    react-js,
+    state-management,
+    redux,
+    introduction,
+  ]
+---
+
+## Introduction
+
+Redux Toolkit is the official, recommended way to write Redux logic. It provides a set of tools and best practices that make it easier to write Redux applications. Redux Toolkit includes utilities to simplify common use cases, such as store setup, creating reducers, and writing immutable update logic.
+
+## Why Use Redux Toolkit?
+
+Redux Toolkit addresses many common concerns when working with Redux:
+
+1. **Boilerplate Reduction:** Simplifies the process of creating reducers, actions, and the store.
+2. **Immutability:** Enforces immutable state updates using Immer.
+3. **Efficiency:** Includes performance optimizations and simplifies common tasks.
+4. **Configuration:** Provides built-in middleware like Redux Thunk for handling side effects.
+
+## Getting Started with Redux Toolkit
+
+### 1. Installing Redux Toolkit
+
+To get started with Redux Toolkit, you need to install it along with React-Redux:
+
+```bash
+npm install @reduxjs/toolkit react-redux
+```
+
+### 2. Creating a Redux Store
+
+Redux Toolkit provides a `configureStore` function that simplifies store creation. It includes good defaults and automatically sets up the Redux DevTools extension.
+
+```javascript
+// src/app/store.js
```

Comment:
remove `// src/app/store.js` and use like 

````md
```Javascript title="src/app/store.js"
// your code here
```
````

Reply:
and apply for all code